src/components/VideoCard/VideoCard.js

```diff
@@ -0,0 +1,62 @@
+import styles from "./VideoCard.module.css";
+import { Link } from "react-router-dom";
+import generateThumbnail from "../../utils/generateThumbnail";
+import { BsThreeDotsVertical } from "react-icons/bs";
+import { useAuth } from "../../contexts/AuthProvider";
+import { useHistory } from "../../contexts/HistoryProvider";
+
+const VideoCard = ({ video, onOptionClick }) => {
+  const { token } = useAuth();
+  const { addToHistory } = useHistory();
+
+  const handleVideoClick = async (videoId) => {
+    await addToHistory(videoId);
+  };
+
+  return (
+    <div className={styles.videoItem}>
+      <Link
+        onClick={() => handleVideoClick(video._id)}
+        to={`/video/${video._id}`}
```

Comment:
Why onClick on `<Link>`? Try to avoid that.

See what happens:
Link onClick will invoke the Promise and route to "/video/${video._id}", so this VideoCard component unmounts but the Promise will be still resolving. So it will work but it's an anti-pattern (running something of an unmounted component).

Why not add this `addToHistory` on that videoPlayer components useEffect, I guess that makes more sense!